### PR TITLE
Make it possible to configure KAFKA_JVM_PERFORMANCE_OPTS

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- ~~Bugfix~~
- Enhancement / new feature
- ~~Refactoring~~

### Description

This PR makes it possible to configure some additional options of the JVM - mainly related to performance. These options are passed to KAfka, Zoo and Connect through `KAFKA_JVM_PERFORMANCE_OPTS` environment variable.

Additionally it renames the method `kafkaHeapOptions` to `heapOptions` to avoid thew confusion that this method is used only for Kafka and not for other assemblies.

This relates to the discussion in #441.

As a part of this PR I also enhanced the system tests. I did also enabled usage of ephemeral storage for all Kafka Connect tests and for the JVM options tests in Kafka. That should give us some speedup with not sacrificing any quality of the system tests.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java [hunk 1]
```diff
@@ -63,4 +64,40 @@ public void testJvmMemoryOptionsMemoryRequest() {
...
+
+    @Test
+    public void testJvmPerformanceOptions() {
+        JvmOptions opts = JvmOptions.fromJson("{}");
+
+        assertNull(getPerformanceOptions(opts));
+
+        opts = JvmOptions.fromJson("{\n" +
+                "  \"-server\": \"true\"\n" +
+                "}");
```
Comment (ppatierno) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java]: why not just one line for this simple JSON ?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/JvmOptionsTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        opts = JvmOptions.fromJson("{\n" +
+                "  \"-server\": true\n" +
+                "}");
+
+        assertTrue(opts.getServer());
+
+        opts = JvmOptions.fromJson("{\n" +
+                "  \"-server\": \"false\"\n" +
+                "}");
```
Comment (ppatierno) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/JvmOptionsTest.java]: one line for this simple JSON ? The same for the others above ?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 3]
```diff
@@ -650,7 +651,12 @@ public void setJvmOptions(JvmOptions jvmOptions) {
         this.jvmOptions = jvmOptions;
     }
 
-    protected void kafkaHeapOptions(List<EnvVar> envVars, double dynamicHeapFraction, long dynamicHeapMaxBytes) {
+    /**
+     * Adds KAFKA_HEAP_OPTS variable to the EnvVar list if any heap related options were specified.
+     *
+     * @param envVars List of Environment Variables
+     */
+    protected void heapOptions(List<EnvVar> envVars, double dynamicHeapFraction, long dynamicHeapMaxBytes) {
```
[the remark this answers is not in the document]
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: `KAFKA_HEAP_OPTS` is the name of the environment variable accepted by the `kafka-run-class.sh` script. That is just copy paste from Kafka code. The method is our own method. If everyone insists, I can rename it back. But I spent 5 minutes looking for zookeperHeapOptions before I decided to rename it. So there comes my motivation.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java]: Might be subjective, but for me it is much easier to read and edit like this. The only place where I used one line JSON was when adding something to existing one line JSONs.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/JvmOptionsTest.java (hunk 2 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/JvmOptionsTest.java]: Might be subjective, but for me it is much easier to read and edit like this. The only place where I used one line JSON was when adding something to existing one line JSONs.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java (hunk 1 above)
Comment (ppatierno) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/model/AbstractModelTest.java]: Yeah it's subjective but maybe you don't really need the "\n" which just add noise. Without "\n" is should be still simple to read for you from a developer perspective I guess. Btw up to you.

File: documentation/book/cluster-operator.adoc [hunk 4]
```diff
@@ -485,6 +485,38 @@ suffix. This is in contrast to the units used for <<resources_json_config,memory
...
+----
+
+The example configuration above will result in following JVM options:
+
+[source]
+----
+-server -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35 -XX:+ExplicitGCInvokesConcurrent -XX:-UseParNewGC
+----
+
+When none of the two options (`-server` and `-XX`) is specified, the default Apache Kafka configuration of `KAFKA_JVM_PERFORMANCE_OPTS` will be used.
```
Comment (kyguy) [inline documentation/book/cluster-operator.adoc]: none -> neither

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java]: Got it ... the problem was having the KAFKA prefix but now from the Kafka source code I see that these variables KAFKA_XXXXX are used for Zookeeper as well which I forgot uses the same `kafka-run-class.sh` script.

File: documentation/book/cluster-operator.adoc [hunk 5]
```diff
@@ -485,6 +485,38 @@ suffix. This is in contrast to the units used for <<resources_json_config,memory
...
+             "UseG1GC": true,
+             "MaxGCPauseMillis": 20,
+             "InitiatingHeapOccupancyPercent": 35,
+             "ExplicitGCInvokesConcurrent": true,
+             "UseParNewGC": false
+         }
+}
+----
+
+The example configuration above will result in following JVM options:
```
Comment (kyguy) [inline documentation/book/cluster-operator.adoc]: in following -> in _the_ following

Comment (scholzj) [reply documentation/book/cluster-operator.adoc]: Stupid mistake ... I'm getting these ever PR :-(

**Pull request merged**